### DISCUSSION TITLE
Data fetching in Next Component

### DISCUSSION THREAD

**Discussion by H3lio2f** [vercel/next.js#22652] [Help]
Good night or good afternoon guys 🤗

 I have some concerns about fetching data in Nextjs! 

I can do a data fetching in a NextPage using getServerSidePros, getStaticProps, 🤗 but in components I can't, even using useEffect Hooks 😩 

Since I can't use getServerSidePros, getStaticProps on components, what is the correct approach for fetching data?

Supposing we want to fetch data in a component that is not nested on any page.

 🤔 I would appreciate it 🙏

**Comment by indegser** [comment]
I think `swr` is best fit for Next.js in terms of data fetching 😍

```tsx
const { data } = useSWR('apikey', () => axios.get('https://...') )
```
